### PR TITLE
Avoid undefined behaviour due to division by zero

### PR DESCRIPTION
**Pull request by andreas-schwab**

### CODE REVIEW

**Comment by hugovk** [comment]
@andreas-schwab Is this testable via a unit test?

**Comment by andreas-schwab** [comment]
On Jun 30 2018, Hugo <notifications@github.com> wrote:

> @andreas-schwab Is this testable via a unit test?

It fixes the testsuite.

Andreas.

-- 
Andreas Schwab, schwab@linux-m68k.org
GPG Key fingerprint = 7578 EB47 D4E5 4D69 2510  2552 DF73 E780 A9DA AEC1
"And now for something completely different."

File: src/libImaging/QuantOctree.c [hunk 1]
```diff
@@ -149,10 +149,17 @@ count_used_color_buckets(const ColorCube cube) {
 static void
 avg_color_from_color_bucket(const ColorBucket bucket, Pixel *dst) {
    float count = bucket->count;
-   dst->c.r = (int)(bucket->r / count);
-   dst->c.g = (int)(bucket->g / count);
-   dst->c.b = (int)(bucket->b / count);
-   dst->c.a = (int)(bucket->a / count);
+   if (bucket->count != 0) {
```
Comment (radarhere) [inline src/libImaging/QuantOctree.c]: Minor: I suggest that this line would be better as `if (count != 0) {` - that is variable that we are concerned with in the next block.

**Comment by andreas-schwab** [comment]
On Jun 30 2018, Andrew Murray <notifications@github.com> wrote:

> radarhere commented on this pull request.
>
>
>
>> @@ -149,10 +149,17 @@ count_used_color_buckets(const ColorCube cube) {
>  static void
>  avg_color_from_color_bucket(const ColorBucket bucket, Pixel *dst) {
>     float count = bucket->count;
> -   dst->c.r = (int)(bucket->r / count);
> -   dst->c.g = (int)(bucket->g / count);
> -   dst->c.b = (int)(bucket->b / count);
> -   dst->c.a = (int)(bucket->a / count);
> +   if (bucket->count != 0) {
>
> Minor: I suggest that this line would be better as `if (count != 0) {` - that is variable that we are concerned with in the next block.

That would change an integer compare into a float compare.

Andreas.

-- 
Andreas Schwab, schwab@linux-m68k.org
GPG Key fingerprint = 7578 EB47 D4E5 4D69 2510  2552 DF73 E780 A9DA AEC1
"And now for something completely different."

**Comment by andreas-schwab** [comment]
https://build.opensuse.org/package/live_build_log/openSUSE:Factory:RISCV/python-Pillow/standard/riscv64

[  668s] =================================== FAILURES ===================================
[  668s] __________________ TestFilePng.test_save_p_transparent_black ___________________
[  668s] 
[  668s] self = <test_file_png.TestFilePng testMethod=test_save_p_transparent_black>
[  668s] 
[  668s]     def test_save_p_transparent_black(self):
[  668s]         # check if solid black image with full transparency
[  668s]         # is supported (check for #1838)
[  668s]         im = Image.new("RGBA", (10, 10), (0, 0, 0, 0))
[  668s]         self.assertEqual(im.getcolors(), [(100, (0, 0, 0, 0))])
[  668s]     
[  668s]         im = im.convert("P")
[  668s]         test_file = self.tempfile("temp.png")
[  668s]         im.save(test_file)
[  668s]     
[  668s]         # check if saved image contains same transparency
[  668s]         im = Image.open(test_file)
[  668s] >       self.assertEqual(len(im.info["transparency"]), 256)
[  668s] E       TypeError: object of type 'int' has no len()
[  668s] 
[  668s] Tests/test_file_png.py:288: TypeError
[  668s] ============== 1 failed, 355 passed, 26 skipped in 62.71 seconds ===============
[

**Comment by hugovk** [comment]
Merged in https://github.com/python-pillow/Pillow/pull/3388.

**Pull request closed without merging**